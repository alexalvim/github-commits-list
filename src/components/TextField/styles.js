import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

const Input = styled.input`
  background-color: ${Colors.lightestColor};
  color: ${Colors.darkestColor}
  font-size: ${Typo.medium};
  padding: ${Spaces.half};

  ${(props) => props.wrapperStyle}
`;

export {
  Input
};