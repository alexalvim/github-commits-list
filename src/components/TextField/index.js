import React from 'react';

import { Input } from './styles'

export default ({ value, onChange, placeholder, wrapperStyle }) => (
  <Input
    value={value}
    onChange={onChange}
    placeholder={placeholder}
    wrapperStyle={wrapperStyle}/>
)