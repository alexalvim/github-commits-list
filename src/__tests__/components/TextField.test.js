import React from 'react';
import { shallow, mount } from 'enzyme';

import TextField from '../../components/TextField';

describe('Testing TextField', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<TextField value='' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('call expected function', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<TextField value='' onChange={mockFunction} />);

    wrapper.simulate('change', {target: {value: 'a'}});
    expect(mockFunction).toHaveBeenCalled();    
  }) 
});