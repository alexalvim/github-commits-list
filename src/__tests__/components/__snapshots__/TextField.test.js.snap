// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextField should render correctly 1`] = `ShallowWrapper {}`;
